datastore/src/test/java/io/spine/server/storage/datastore/DsShardedWorkRegistryTest.java

```diff
@@ -0,0 +1,153 @@
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.server.storage.datastore;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.testing.NullPointerTester;
+import com.google.protobuf.Timestamp;
+import com.google.protobuf.util.Timestamps;
+import io.spine.base.Identifier;
+import io.spine.server.NodeId;
+import io.spine.server.delivery.ShardIndex;
+import io.spine.server.delivery.ShardProcessingSession;
+import io.spine.server.delivery.ShardSessionRecord;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import java.util.Optional;
+
+import static com.google.common.truth.Truth.assertThat;
+import static io.spine.server.storage.datastore.given.TestShardIndex.newIndex;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+@DisplayName("DsShardedWorkRegistry should")
+public class DsShardedWorkRegistryTest {
+
+    private static final ShardIndex index = newIndex(1, 15);
+
+    private static final NodeId nodeId = newNode();
+
+    private final TestDatastoreStorageFactory factory =
+            TestDatastoreStorageFactory.defaultInstance();
+
+    private DsShardedWorkRegistry registry;
+
+    @BeforeEach
+    void setUp() {
+        factory.setUp();
+        registry = new DsShardedWorkRegistry(factory);
+    }
+
+    @AfterEach
+    void tearDown() {
+        factory.tearDown();
+    }
+
+    @Test
+    @DisplayName("pick up the shard and write a corresponding record to the storage")
+    public void pickUp() {
+        Optional<ShardProcessingSession> session = registry.pickUp(index, nodeId);
+        assertTrue(session.isPresent());
+        assertThat(session.get()
+                          .shardIndex()).isEqualTo(index);
+
+        ShardSessionRecord record = readSingleRecord(index);
+        assertThat(record.getIndex()).isEqualTo(index);
+        assertThat(record.getPickedBy()).isEqualTo(nodeId);
+    }
+
+    @Test
+    @DisplayName("not be able to pick up the shard if it's already picked up")
+    public void cannotPickUpIfTaken() {
+
+        Optional<ShardProcessingSession> session = registry.pickUp(index, nodeId);
+        assertTrue(session.isPresent());
```

Comment:
`assertThat(session).isPresent()` reduces chances of developing strabismus. Please consider.

Reply:
I have issues with several `assertThat(..)` static methods which do not belong to the same class:
* `Truth.assertThat(..)`.
* `Truth8.assertThat(..)`.
* `Assert.assertThat(..)` (the JUnit 5).

Even if I am not using the latter one, I am still not able to statically import them both, so one of the calls will stay as `Class.methodCall()`. That makes me stumble upon the method choice while writing the code every time. Not sure if this is better than the chances of developing strabismus.